openfl/component/aggregator/aggregator.py

```diff
@@ -306,21 +306,18 @@ def get_tasks(self, collaborator_name):
                 t for t in tasks if not self._collaborator_task_completed(
                     collaborator_name, t, self.round_number)
             ]
-            for t in tasks:
-                if t in self.stragglers_for_task:
-                    for stragglers in self.stragglers_for_task[t]:
-                        if collaborator_name in stragglers:
-                            tasks.remove(t)
+            for straggler in self.stragglers:
+                if collaborator_name == straggler:
+                    tasks = []
+
         else:
             tasks = [
                 t for t in tasks if not self._collaborator_task_completed(
                     collaborator_name, t.name, self.round_number)
             ]
-            for t in tasks:
-                if t.name in self.stragglers_for_task:
-                    for stragglers in self.stragglers_for_task[t.name]:
-                        if collaborator_name in stragglers:
-                            tasks.remove(t)
+            for straggler in self.stragglers:
+                if collaborator_name == straggler:
```

Comment:
This could look simpler
```suggestion
            if collaborator_name in self.stragglers: